pkg/scheduler/scheduler.go

```diff
@@ -552,7 +552,9 @@ func (sched *Scheduler) scheduleOne() {
 			if forgetErr := sched.Cache().ForgetPod(assumedPod); forgetErr != nil {
 				klog.Errorf("scheduler cache ForgetPod failed: %v", forgetErr)
 			}
-			sched.recordSchedulingFailure(assumedPod, permitStatus.AsError(), reason, permitStatus.Message())
```

Comment:
Why not place this line before `return`?

Reply:
@draveness 
`defer sched.recordSchedulingFailure(assumedPod, permitStatus.AsError(), reason, permitStatus.Message())` 
not equal to
`defer func() {
sched.recordSchedulingFailure(assumedPod, permitStatus.AsError(), reason, permitStatus.Message())
			}()`

First will execute immediately `sched.recordSchedulingFailure(assumedPod, permitStatus.AsError(), reason, permitStatus.Message())`

Reply:
@wgliang do you prefer to use defer or change the sequence of these two lines ?

Reply:
> `defer func() { sched.recordSchedulingFailure(assumedPod, permitStatus.AsError(), reason, permitStatus.Message()) }()`

I know the difference, just want to ensure it is on purpose.

Reply:
> @wgliang do you prefer to use defer or change the sequence of these two lines ?

@danielqsj I'm ok with that,but I want to know if make test case passed is a coincidence?

Reply:
I test it several times, all passed.

Reply:
I prefer to just call recordSchedulingFailure at the end before the return rather than doing the defer to simplify the code.